happ/zomes/profiles/src/lib.rs

```diff
@@ -192,12 +205,30 @@ pub fn fetch_agents(_: ()) -> ExternResult<Vec<Profile>> {
             GetOptions::content(),
         )?
         .into_iter()
-        // TODO: this should deduplicate results where the .agent_pub_key
-        // is the same. It should be specific and pick/prefer one where
-        // profile.is_imported is `false`.
         .map(|wire_record| wire_record.entry)
         .collect();
-    Ok(entries)
+    
+    // use a BTreeMap to associate a profile struct to an agent pub key, used to dedup profiles for the same agent
+    let mut unique_profiles: BTreeMap<AgentPubKeyB64, Profile> = BTreeMap::new();
+    let _ = entries.into_iter()
+        .map(|profile| {
```

Comment:
Instead of .map make this one a for loop, as it is performing a side effect not a transformation